modules/gapi/src/backends/onnx/gonnxbackend.cpp

```diff
@@ -590,13 +636,32 @@ cv::GMatDesc ONNXCompiled::outMeta(int idx) const {
                         toCV(out_tensor_info[ort_idx].dims));
 }
 
-void ONNXCompiled::setInput(int i, const cv::Mat &m) {
-    const auto in_idx  = i;
+void ONNXCompiled::setInput(int in_idx, const cv::Mat &m) {
+    GAPI_Assert(!m.empty() && "Input data can't be empty!");
     const auto in_name = params.input_names[in_idx];
     const auto ort_idx = getIdxByName(in_tensor_info, in_name);
     preprocess(m, in_tensor_info[ort_idx], in_data[in_idx]);
 }
 
+void ONNXCompiled::extractMat(ONNXCallContext &ctx, const size_t in_idx, Views& views) {
+    switch (ctx.in_shapes[in_idx]) {
+        case cv::GShape::GFRAME: {
+            const cv::MediaFrame& frame = ctx.inFrame(in_idx);
+            views.emplace_back(new cv::MediaFrame::View(frame.access(cv::MediaFrame::Access::R)));
+            GAPI_Assert(views.size() <= numInputs());
+            preprocess(*views.back(), frame.desc(), exMat);
+            break;
+        }
+        case cv::GShape::GMAT: {
+            exMat = ctx.inMat(in_idx);
```

Comment:
It also requires preprocessing, isn't it?

Reply:
...resize, normalization, etc. Where does it happen now for a regular GMat input?

Reply:
oh I see you do it also in `setInput()`. This flow is somewhat complicated, do you have two `preprocess()` functions now which do different things?